internal/vendors/service_test.go

```diff
@@ -52,3 +56,71 @@ func TestVendorService_GetSomeStuff(t *testing.T) {
 		})
 	}
 }
+
+func TestVendorService_GetAll(t *testing.T) {
+	sampleData := []Vendor{
+		{
+			ID:            "1",
+			Name:          "name",
+			Description:   "description",
+			BpId:          1,
+			BpName:        "bp_name",
+			Rating:        1,
+			AreaGroupId:   1,
+			AreaGroupName: "group_name",
+			SapCode:       "sap_code",
+			ModifiedDate:  time.Now(),
+			ModifiedBy:    1,
+			Date:          time.Now(),
+		},
+	}
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	type fields struct {
+		mockVendorDBAccessor *MockvendorDBAccessor
+	}
+	type args struct {
+		ctx context.Context
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    []Vendor
+		wantErr bool
+	}{
+		{
+			name: "success",
+			fields: fields{
+				mockVendorDBAccessor: NewMockvendorDBAccessor(ctrl),
+			},
+			args:    args{ctx: context.Background()},
+			want:    sampleData,
+			wantErr: false,
+		},
+		{
+			name: "failure",
+			fields: fields{
+				mockVendorDBAccessor: NewMockvendorDBAccessor(ctrl),
+			},
+			args:    args{ctx: context.Background()},
+			want:    nil,
+			wantErr: true,
```

Comment:
Sorry, I just noticed. Is `wantErr` expected not to be used? I though it should we used below in the for loop to throw some error if it's true (?) 

cc: @valeelim